### PR TITLE
Add save command.

### PR DESCRIPTION
**Pull request by fennecdjay**
Here it is ;-)
It looks just fine to me.
---------

### CODE REVIEW

**Comment by fennecdjay** [comment]
We're almost done!
I'd just like rmdir_r() to remove user directory if it is empty.

**Comment by desyncr** [comment]
Nice! I'm gonna take a look tonight!

File: README.md [hunk 1]
```diff
@@ -55,6 +55,26 @@ zpm update
 
 ```
 
+To remove plugin:
+
+```sh
+zpm remove user/plugin
```
Comment (desyncr) [inline README.md]: Not actually necessary but to keep a unified syntax on the README.md and in general, it should quote the plugin name. ie: `zpm remove "user/plugin"`

File: zpm.c [hunk 2]
```diff
@@ -399,6 +500,21 @@ int plugin_print_list() {
...
 
+int plugin_print_script() {
+    char entry[PATH_MAX];
+    char* zpm_init = get_plugin_list_path();
+    FILE* store = fopen(zpm_init, "r");
+
+    printf("#!/bin/sh\n");
+    memset(entry, 9, PATH_MAX);
+    while( fgets(entry, PATH_MAX, store)) {
+        printf("zpm %s", entry);
```
Comment (desyncr) [inline zpm.c]: Will need quotes on plugin name.

File: README.md [hunk 3]
```diff
@@ -55,6 +55,26 @@ zpm update
...
+
+```sh
+zpm uninstall user/plugin
+zpm uninstall /path/to/local/plugin
+```
+
+To make a copy of your config
+
+```sh
+zpm save > .zpmrc
```
Comment (desyncr) [inline README.md:75]: We probably will need to explain the workflow with an example. Ex:

Install some plugins:

    zpm "some/plugin"
    zpm "other/plugin"

Then you can persist your current configuration to be able to commit it into a dotfile repository:

    zpm save > ~/dotfiles/.zpmrc

Next time you'll only need to install them like so:

    source ~/dotfiles/.zpmrc

**Review by desyncr — CHANGES_REQUESTED** [review]
Some minor changes.

File: zpm.c [hunk 4]
```diff
@@ -399,6 +500,23 @@ int plugin_print_list() {
     return 0;
 }
 
+int plugin_print_script() {
+    char entry[PATH_MAX];
+    char* zpm_init = get_plugin_list_path();
+    FILE* store = fopen(zpm_init, "r");
+
+    printf("#!/bin/sh\n");
```
Comment (desyncr) [inline zpm.c]: I believe there is no need for this.

Comment (fennecdjay) [reply zpm.c]: Right. I just tought it was more correct.
Removing it.

**Comment by fennecdjay** [comment]
Maybe you should rework the README.md, as you clearly know zpm's workflow better than I do.

**Comment by desyncr** [comment]
Seems fair. I'll create another issue for the README.md

**Comment by desyncr** [comment]
Last thing would be to add `remove` and `uninstall` to the list of available commands.

**Comment by desyncr** [comment]
On a quick test it failed to `uninstall` the plugin directory. It seems it removed the `.git` directory though.

I also believe we could change terms: "remove" for unlinking/remove directory, and "disable" to remove plugin from listing. It's a small thing but it's best to get it correct the first time.

**Comment by fennecdjay** [comment]
If you agree, I'll comment everything in rmdir_r(), and just use something like
```c
system("rm -r xxx")
```
so we get the functionnality, until I fix that function.

**Comment by desyncr** [comment]
@fennecdjay I do agree. For now it'll suffice.

**Comment by fennecdjay** [comment]
We can ignore Travis warning for now, as it complains about rake (?).
On my station it seems to work just fine.

**Comment by fennecdjay** [comment]
Also, I see you created zpm-project organisation, so I suppose you expect next PR to be [there](https://github.com/zpm-project/zpm-zsh).

**Comment by fennecdjay** [comment]
Oups...
Pushing the rigth thing now.

**Comment by desyncr** [comment]
@fennecdjay There is a lot of comments and useful interactions, so let's keep it here until the final approval. Then we'll move to zpm-project. Sorry for the mess up but it has to be sooner or later :)

**Comment by fennecdjay** [comment]
Very good to me. ;-)

File: zpm.c [hunk 5]
```diff
@@ -217,6 +217,118 @@ int mkdir_p(const char *path) {
     return 0;
 }
 
+int rmdir_r(const char *path) {
+/*
```
Comment (desyncr) [inline zpm.c]: Remove all this commented out code.

File: zpm.t [hunk 6]
```diff
@@ -31,8 +31,8 @@ Subsequentially add a new item to the list.
 List command show list of installed plugins.
 
     $ ./zpm list
-    zsh-users/zsh-syntax-highlighting
-    rupa/z
+    zsh-users/zsh-syntax-highlighting .*
```
Comment (desyncr) [inline zpm.t]: We should probably add tests for this but for now I want to get this merged ASAP to avoid huge git conflicts.

**Review by desyncr — CHANGES_REQUESTED** [review]
One last thing.

**Comment by desyncr** [comment]
One last thing then we can merge this. Once done you have to create a new PR in `zpm-project/zpm-zsh` repository.

File: zpm.t (hunk 6 above)
Comment (fennecdjay) [reply zpm.t]: Fine. reverting to master version for now.

**Comment by fennecdjay** [comment]
I'm waiting for your signal to either do more changes or create PR in zpm-project/zpm-zsh repository.

**Review by desyncr — APPROVED** [review]
We are done! Great job! Let's move it to `zpm-project/zpm-zsh` repository.

**Comment by fennecdjay** [comment]
Great! doing it rigth now.

**Comment by desyncr** [comment]
Also, remember to review https://github.com/zpm-project/zpm-zsh/pull/6

**Comment by fennecdjay** [comment]
Sure thing!
Yet I have to figure out how to do so ;-)

**Comment by desyncr** [comment]
Moved to https://github.com/zpm-project/zpm-zsh/pull/12

**Pull request closed without merging**